### DISCUSSION TITLE
How to collaborate, i.e. share Storage?

### DISCUSSION THREAD

**Discussion by kevin-hanselman** [amakelov/mandala#12] [General]
Hi, @amakelov. I'm super excited about Mandala! It seems to address some of the biggest pain points I have with many tools in this space, especially DVC and even my own project, [Dud](https://github.com/kevin-hanselman/dud): facilitating iterative development and taming the cognitive load of data/computation sprawl. Additionally, your tutorial notebooks are very helpful in quickly getting up to speed on what Mandala does; thanks for writing them!

What's Mandala's approach and/or vision for collaboration? For example, say you have your model trained, you're happy with its performance, and you want to share your code and model with colleagues; how do you do this? I see you have some code (in mandala) for remote storage options; is this the route you expect teams to take when developing with Mandala? How does Mandala-tracked code/data interface with Git and a Git provider (e.g. GitHub)?

Thanks again for creating Mandala! I plan to kick the tires soon to better understand what's happening under the hood.

**Comment by amakelov** [comment]
Hi @kevin-hanselman and thank you for your interested in the project! 

There's currently no support for what you are describing, though you are correct at one point this was possible using the remote storage. To get collaboration of this kind to work, there are a few major missing pieces:
- the remote storage should represent the versioning information for the source code (currently this is completely missing);
- logic to capture the python environment used by a storage, so that we can check that another person using another environment will be able to use the storage correctly. This is quite open-ended and potentially messy, but a somewhat hacky first approach that I expect to work well for most cases would be to just check the semantic versions of all packages and flag any breaking changes between the two environments.

I have no capacity (or use cases) in the near future to work on this, though it could be a valuable addition eventually.

Always happy to hear suggestions / get pull requests / collaborate to make this (and other parts of the library) better!
